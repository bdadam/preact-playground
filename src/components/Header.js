/** @jsx h */

import { h, Component } from 'preact';

import './Header.scss';


export default class Header extends Component {

    constructor() {
        super();
        this.state.open = false;
    }
    
    render(props, state) {

        return (
            <header class="site-header">
                <div class="site-header__header-bar">
                    <button class="site-header__menu-toggle" onClick={this.toggleMenu.bind(this)}>Menu</button>
                    Site Header
                </div>
                {
                    state.open ? (
                        <nav class="site-header__nav">
                            <a href="/">Home</a>
                            <a href="/">About</a>
                        </nav>) : ''
                }
            </header>
        );
    }

    toggleMenu(a,b) {
        const oldState = this.state.open;
        this.setState({ open: !oldState });
        console.log(this.state.open);
    }
}