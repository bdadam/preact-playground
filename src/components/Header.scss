.site-header {
    background-color: #fff;
}

.site-header__header-bar {
    padding: 20px;
    font-size: 1.25rem;
    line-height: 1;
    border-bottom: solid 1px #dcdcdc;
}

.site-header__nav {
    display: none;
    border: 1px solid #ddd;   
    padding: 20px;

    a {
        display: block;
        padding: 0 4px;
    }

    .site-header--open & {
        display: block;
    }
}

.site-header__menu-toggle {
    border: 1px solid #ccc;
    // width: 20px;
    // height: 20px;
    // display: block;
    padding: 12px;
    margin-right: 12px;
    vertical-align: middle;
}