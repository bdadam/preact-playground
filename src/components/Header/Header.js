/** @jsx h */

import classnames from 'classnames';

import { h, Component } from 'preact';

import './Header.scss';

const menuItems = [
    { title: 'Home Title', href: '', text: 'Home' },
    { title: 'About Title', href: '', text: 'About' },
    { title: 'Yaaayy Title', href: '', text: 'Yaaayy' },
];

const cls = { asdf: true, qwertt: 0 };

const Header = ({open, onToggle}) => {
    return (
        <header class={classnames("site-header", { "site-header--open": open })}>
            <div class="site-header__header-bar">
                <button class="site-header__menu-toggle" onClick={onToggle}>Menu</button>
                Site Header
            </div>
            <nav class="site-header__nav">
                { menuItems.map(item => (<a href={item.href} title={item.title}>{item.text}</a>)) }
            </nav>
        </header>
    );
};

export default class extends Component {
    constructor() {
        super();
        this.state.open = false;
    }

    render() {
        return <Header open={this.state.open} onToggle={this.toggle.bind(this)}/>;
    }

    toggle() {
        this.setState({ open: !this.state.open });
    }
}