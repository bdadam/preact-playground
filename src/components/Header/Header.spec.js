import { h } from 'preact';
import render from 'preact-render-to-string';
import htmlLooksLike from 'html-looks-like';

import Header from './Header';

describe('Site Header', () => {

    it('looks like', () => {
        const actual = render(<Header />);
        const expected = `
            <header class="site-header">
                <div class="site-header__header-bar">
                    <button class="site-header__menu-toggle">Menu</button>
                    Site Header
                </div>
                <nav class="site-header__nav">
                    {{ ... }}
                </nav>
            </header>`;

        htmlLooksLike(actual, expected);
    });

    it('matches snapshot', () => {
        const tree = render(<Header />);
        expect(tree).toMatchSnapshot();
    });

    it('opens on button click', () => {
        const header = new Header();

        expect(header.state.open).toBe(false);
        header.toggle();
        expect(header.state.open).toBe(true);
        header.toggle();
        expect(header.state.open).toBe(false);
    });
});