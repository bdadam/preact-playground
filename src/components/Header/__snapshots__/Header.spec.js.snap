// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Site Header matches snapshot 1`] = `
<header class="site-header">
  <div class="site-header__header-bar">
    <button class="site-header__menu-toggle">Menu</button>Site Header</div>
  <nav class="site-header__nav">
    <a href title="Home Title">Home</a>
    <a href title="About Title">About</a>
    <a href title="Yaaayy Title">Yaaayy</a>
  </nav>
</header>
`;
